File: RetailStation.Entities/DTOs/SystemSettings/CountryModel.cs
using RetailStation.Entities.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.DTOs.SystemSettings
{
    public class CountryModel : CreatorModel
    {
        public int? CountryId { get; set; }

        public string NameAR { get; set; }


        public string NameEN { get; set; }

        public bool IsActive { get; set; } = true;
        public int? TotalCount { get; set; }
    }
    public class CityModel : CreatorModel
    {

        public int? CityId { get; set; }


        public int CountryId { get; set; }


        public string PostalCode { get; set; }


        public string NameAR { get; set; }


        public string NameEN { get; set; }

        public bool IsActive { get; set; } = true;
        public int? TotalCount { get; set; }
    }
    public class RegionModel : CreatorModel
    {

        public int? RegionId { get; set; }


        public string Code { get; set; }


        public string NameAR { get; set; }


        public string NameEN { get; set; }

        public int? CityId { get; set; }
        public int? CountryId { get; set; }
        public bool IsActive { get; set; } = true;

        public int? TotalCount { get; set; }

    }
}

File: RetailStation.Angular/src/app/components/Admin/models/Operation/CountryModel.ts
import { CreatorModel } from 'src/app/components/Shared/models/CreatorModel';
export interface CountryModel extends CreatorModel {
  countryId: number;
  nameAR: string;
  nameEN: string;
  isActive: boolean;
}


export interface CityModel  extends CreatorModel{
  cityId: number;
  countryId: number;
  postalCode?: string;
  nameAR: string;
  nameEN: string;
  isActive: boolean;
  
}

export interface RegionModel extends CreatorModel {
  regionId: number;
  code?: string;
  nameAR: string;
  nameEN: string;
  cityId?: number;
  countryId?: number;
  isActive: boolean;

}

